service/src/test/java/org/sunbird/learner/actors/bulkupload/TextbookTocActorTest.java

```diff
@@ -24,145 +31,219 @@
+    if (error) {
+      ProjectCommonException res =
+          probe.expectMsgClass(duration("10 second"), ProjectCommonException.class);
+      return res;
+    }
+    Response res = probe.expectMsgClass(duration("10 second"), Response.class);
+    return res;
   }
 
-  @Ignore
-  public void testTocUplodSuccess() throws IOException {
-    TestKit probe = new TestKit(system);
-    ActorRef toc = system.actorOf(props);
-
-    Response readTb = getTbWithoutChildren();
-    when(TextBookTocUtil.readContent(Mockito.anyString())).thenReturn(readTb);
+  private void mock(boolean bool) {
+    when(TextBookTocUtil.getRelatedFrameworkById(Mockito.anyString()))
+        .thenReturn(getFrameworkMap());
+    when(TextBookTocUtil.readHierarchy(Mockito.anyString())).thenReturn(getReadHierarchy(bool));
+    when(TextBookTocUtil.readContent(Mockito.anyString())).thenReturn(getReadContentTextbookData());
+  }
 
-    Request request = new Request();
-    request.setContext(
+  private Response getReadHierarchy(boolean error) {
+    // TODO Auto-generated method stub
+    Response res = new Response();
+    List<String> dialCodes = new ArrayList<>();
+    if (error) {
+      dialCodes.add("1564");
+    } else {
+      dialCodes.add("2019");
+    }
+    res.put(
+        JsonKey.CONTENT,
         new HashMap<String, Object>() {
           {
-            put("userId", "test");
+            put(JsonKey.DIAL_CODES, dialCodes);
+            put(JsonKey.CHILDREN, new ArrayList<>());
+            put(JsonKey.IDENTIFIER, "do_1126788813057638401122");
           }
         });
-    request.put(JsonKey.TEXTBOOK_ID, "do_11265332762881228812868");
-    request.put(JsonKey.DATA, mapper.readValue(TB_CREATE_VALID_REQ, Map.class));
-    request.setOperation(TextbookActorOperation.TEXTBOOK_TOC_UPLOAD.getValue());
-
-    toc.tell(request, probe.getRef());
-    Response res = probe.expectMsgClass(duration("100 second"), Response.class);
-    Assert.assertNotNull(res);
-    Assert.assertEquals("OK", res.getResponseCode().name());
+    return res;
   }
 
-  private static Response getTbWithoutChildren() throws IOException {
-    return mapper.readValue(TB_WITHOUT_CHILDREN, Response.class);
+  private Response getReadContentTextbookData() {
+    Response res = new Response();
+    Map<String, Object> textBookdata = new HashMap<>();
+    List<String> s = new ArrayList<>();
+    s.add("2019");
+    textBookdata.put(JsonKey.RESERVED_DIAL_CODES, s);
+    textBookdata.put(JsonKey.CONTENT_TYPE, CONTENT_TYPE);
+    textBookdata.put(JsonKey.MIME_TYPE, "application/vnd.ekstep.content-collection");
+    textBookdata.put(JsonKey.NAME, "test");
+    res.put(JsonKey.CONTENT, textBookdata);
+    return res;
   }
 
-  private static Response getSuccess() {
-    Response response = new Response();
-    response.put("content_id", "do_11265332762881228812868");
-    return response;
+  private Response getFrameworkMap() {
+    Response res = new Response();
+    return res;
```

Comment:
self-review ur code - first - this metthod is not even required  if it only creates a response object and returns it.